packages/openapi-v3/src/generate-schema.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/openapi-v3
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {SchemaObject} from '@loopback/openapi-v3-types';
+
+interface TypeAndFormat {
+  type?: string;
+  format?: string;
+}
+/**
+ * Generate the `type` and `format` property in a Schema Object according to a
+ * parameter's type.
+ * `type` and `format` will be preserved if provided in `schema`
+ *
+ * @param type The JavaScript type of a parameter
+ * @param schema The schema object provided in an parameter object
+ */
+export function getSchemaForParam(
+  type: Function,
+  schema: SchemaObject,
```

Comment:
Can this be optional?

Reply:
nope, see the method that calls it:

```ts
paramSpec.schema = getSchemaForParam(paramType, paramSpec.schema || {});
```

Reply:
that could be handled within the function itself can't it? something like this:
```
export function getSchemaForparam(
  type: Function,
  schema: SchemaObject,
) {
  schema = schema || {};
  // .. rest of the code here
}
```

Reply:
Good point!